### PR TITLE
Proposed behaviour for RepositorySelector #732

### LINKED ISSUE #732: Proposed behaviour for RepositorySelector
The `RepositorySelector` should reflect the `.json` files saved on disk. 
The list refers to the `comboBox` in `RepositorySelector`. 
This means that: 
1. Opening a new repo from the `RepositorySelector` adds it to the list (last entered repository is the new primary repository). 
2. Opening a new repo from any `FilterTextField` adds it to the list (but doesn't make it the new primary repository). 
3. At startup, the `RepositorySelector` will be populated based on the `.json` files on disk. 
4. We don't need to save a list of last used repositories, just the last valid one that was entered. 
5. To remove a repository from the `RepositorySelector`, the user just has to delete the corresponding `.json` file and restart HT. 
6. If the last valid repository's `.json` file is deleted/missing, we populate the `RepositorySelector` normally but leave the default value blank. (Although this brings about the question of which default repository do we use?)

Alternatively we can save a list of valid last used repositories as a stack so we just go down the list if the last used `.json` is missing. OR we could just pop up the login dialog to force the user to choose a new primary repository. 

This should solve #619 and #665.

### PR DESCRIPTION
**Pull request by lejolly**
Fixes #732, #619 and #665.

### CODE REVIEW

**Comment by damithc** [comment]
test comment. please ignore.

File: src/main/java/ui/GUIController.java [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        HashMap<String, HashSet<Integer>> toUpdate = new HashMap<>();
+
+        // This is not triggered by a (metadata) update, so we pass false into the call.
+        processColumn(changedColumn, multiModel, allModelIssues, toUpdate, false);
+
+        dispatchMetadataRequests(toUpdate);
+    }
+
+    /**
+     * This method manages the flow of execution in filtering and updating a column.
```
Comment (damithc) [inline src/main/java/ui/GUIController.java]: You can omit the `This method/class ...` part. It's ok to start `Manages the flow...`

File: src/main/java/ui/issuepanel/IssuePanel.java [hunk 2]
```diff
@@ -83,19 +83,19 @@ private boolean issueHasNewComments(TurboIssue issue, boolean hasMetadata) {
...
+//
+//        // Only update filter if filter does not contain UPDATED (does not need to wait for metadata)
+//        // or if hasMetadata is true (metadata has arrived), or if getIssueList is empty (if filter does
+//        // have UPDATED, but there are no issues whose metadata require retrieval causing hasMetadata to
+//        // never be true)
+//
+//        if (!hasUpdatedQualifier // not waiting for metadata, just update
+//                || hasMetadata // metadata has arrived, update
+//                || getIssueList().size() == 0 // checked only when above two not satisfied
+//                ) {
```
Comment (damithc) [inline src/main/java/ui/issuepanel/IssuePanel.java]: Remove if not used.

File: src/test/java/guitests/UseGlobalConfigsTest.java [hunk 3]
```diff
@@ -85,10 +86,9 @@ public void globalConfigTest() {
         assertEquals(2, lastOpenFilters.size());
         assertEquals("", lastOpenFilters.get(0));
         assertEquals("repo:dummy2/dummy2", lastOpenFilters.get(1));
-        // Last viewed repositories
-        List<String> lastViewedRepositories = testPref.getLastViewedRepositories();
-        assertEquals("dummy/dummy", lastViewedRepositories.get(0));
-        assertEquals("dummy2/dummy2", lastViewedRepositories.get(1));
+        // Last viewed repository
+        RepositoryId lastViewedRepository = testPref.getLastViewedRepository().get();
+        assertEquals("dummy/dummy", lastViewedRepository.generateId());
```
Comment (damithc) [inline src/test/java/guitests/UseGlobalConfigsTest.java]: Are we testing for all possible cases? There is a lot of functional code and very little test code in this PR.

Comment (lejolly) [inline src/test/java/guitests/UseGlobalConfigsTest.java]: More test code will be added soon.

File: src/main/java/ui/issuepanel/IssuePanel.java (hunk 2 above)
Comment (lejolly) [inline src/main/java/ui/issuepanel/IssuePanel.java]: @acruis will remove it on his pr.

File: src/main/java/ui/GUIController.java (hunk 1 above)
Comment (lejolly) [inline src/main/java/ui/GUIController.java]: @acruis for you to take note.

**Comment by lejolly** [comment]
pr depends on #719 and #746, once they are merged into master, amount of code changes should drop drastically.

File: src/main/java/ui/GUIController.java (hunk 1 above)
Comment (acruis) [inline src/main/java/ui/GUIController.java]: Noted.

**Comment by acruis** [comment]
I'm unclear why my commits are part of this PR.

**Comment by damithc** [comment]
Pbly joshua merged your branch to his branch because his stuff depends on
yours?

**Comment by acruis** [comment]
I think he wanted to make sure that none of his changes clash with mine. 10 points for cautiousness!

**Comment by damithc** [comment]
But it does complicate code reviews, so we have to look for an alternative approach. :-)

**Comment by acruis** [comment]
Should only add repo to list after it has been successfully downloaded.

**Comment by lejolly** [comment]
Noted, will test for that tomorrow.

**Comment by acruis** [comment]
Good to go. Will merge if CI builds.

**Pull request merged**